### ISSUE TITLE
Doesn't seem to be working

### ISSUE THREAD

**Issue by RossBoylan** [jesterKing/literate#25]
Installed the extension in the latest VSCode and could get very little of it to work.

# Current Behavior

1. Created a file `explore.literate`.  This showed `Plain Text` as the mode (bottom right of frame) and none of the markdown was highlighted.
2. `Markdown: Preview` was not available as a command option.
3. `Literate Process` was available.  Running it produced a relatively unformatted but seemingly complete html file (except it didn't show the chunk names) but none of the code files.  I used ```js <<groupb.*>>= explore2.js` as the style for the block open.
4. `Running Extensions` does not show literate as running.
5. `Extension Host` view does not mention it.
6.  `EditorConfig` show stuff like this (node-sqlight-test is my project name):
  ```
node-sqlight-test/explore.literate: {}
node-sqlight-test/explore.literate: {}
node-sqlight-test/explore.literate: {}
node-sqlight-test/explore.literate: {}
exthost: {}
exthost: {}
exthost: {}
node-sqlight-test/explore.literate: {}
undefined: {}
```
I didn't see anything else.

# Previous Behavior/Installation

When I first installed the extension there were more errors and messages showing in the various logs, many looking like 6) above.  `Extension Host` mentioned activating the extension just before stack  traces from (probably) 3 attempts to active the extension. But when I restarted `VSCode`, I got just what's shown above.  I tried disabling and then reenabling the extension, but that didn't seem to produce any debug output at all.

# Speculation
I took a look at the source, but am unfamiliar with the requirements for an extension.  So nothing jumped out at me.  Possibly something about the requirements for extensions has changed since this was written.  Or changes in versions of packages it depends on are creating clashes.  Possibly I have misunderstood something.

In particular, my test file `explore.literate` was trying to see if 2 source files could be generated from it; it wasn't clear to me if that should work.  However, my problems seem to be happening at an earlier step, like the extension isn't getting activated when I open a `.literate` file.

# System Configuration

From `VSCode` Help | About:

Version: 1.84.2 (system setup)
Commit: 1a5daa3a0231a0fbba4f14db7ec463cf99d7768e
Date: 2023-11-09T10:51:52.184Z
Electron: 25.9.2
ElectronBuildId: 24603566
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.17763

I recently installed Node.js 20; maybe there's a clash there.  But I expect VSCode has its own version it uses.

**Comment by jesterKing** [comment]
Hello and thank you for trying out the extension. I'm sorry to hear you're having difficulties using the extension.

I've been busy for the past 1.5 years and not been able to concentrate much on it. I have however been using it with success though on some projects.

I have recently acquired a additional development system, and I have installed the Literate Programming extension via the vscode.

Indeed, at present the extension doesn't know yet how to automatically set .literate files to be handled as markdown files - I have to learn still how that is done. Until then you should be able to make the association for .literate to be handled as markdown though.

This is how it looks with this repository opened:

<img width="1800" alt="image" src="https://github.com/jesterKing/literate/assets/87916/e10e0d52-420f-421f-b27e-bd0a169f4c6b">

The Visual Studio Code version I use:

Version: 1.84.2 (Universal)
Commit: 1a5daa3a0231a0fbba4f14db7ec463cf99d7768e
Date: 2023-11-09T10:52:33.687Z (1 wk ago)
Electron: 25.9.2
ElectronBuildId: 24603566
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 23.1.0

I don't have any other node.js version installed though. I don't know if that could affect the usage of the extension. I would assume that the node.js version bundled with Visual Studio Code is the one being used.

If I figure out more I'll let you know.

BTW, the extension version I am using is v0.10.1

**Comment by RossBoylan** [comment]
Thanks for your response.

I clicked on the bottom right where "Plain Text" showed and changed it to "Markdown", which gets the preview working.  Did you have something more permanent in mind when you said "make the association for .literate to be handled as markdown"?  That sounds more like something that would affect every file with that extension.

Also, I did not realize I needed the `:` between the language type and the fragment name until I saw the sample you put in your response (and noticed the fragment name wasn't in the preview without it).  It might be good to have that in the README.

Finally, when looking at the extension source my preview of `literate.literate` does not have a table of contents, though your screenshot shows one.  Anything I can do about that?

**Comment by jesterKing** [comment]
The README has now a link to https://jesterking.github.io/literate/ . This also explains how the fragments are supposed to be constructed.

The TOC is generated by Markdown AutoTOC extension ( https://marketplace.visualstudio.com/items?itemName=wibblemonkey.markdown-auto-toc ). Install that and you should automatically get a TOC. Eventually I want to add such functionality directly to this extension, though - see #10

**Comment by RossBoylan** [comment]
Thanks for the info.  I still don’t see anything in the README about the `:` after the language specifier at the start of the code fence.

**Comment by jesterKing** [comment]
I'll mention the need for colon in the readme as well: 5eb52bff058b362257fb2d6e55b2a8b87fccd03e

**Comment by jesterKing** [comment]
FWIW I have published v0.11.1 to the market place.

Are you able to work with that and latest main?
